integration/tests/bft/passthrough/dummyOrchestrator.go

```diff
@@ -48,6 +50,9 @@ func NewDummyOrchestrator(logger zerolog.Logger) *dummyOrchestrator {
 // Passing through means that the orchestrator returns the events as they are to the original corrupted nodes so they
 // dispatch them on the Flow network.
 func (d *dummyOrchestrator) HandleEventFromCorruptedNode(event *insecure.Event) error {
+	d.Lock()
```

Comment:
why is a lock on the dummy orchestrator necessary for every event?

Reply:
to mitigate panicing due to [concurrent map access](https://github.com/onflow/flow-go/blob/97a4c6889aa8fb302114f50d5454ed4f53129a68/integration/tests/bft/passthrough/dummyOrchestrator.go#L29).

Reply:
can we add this explanation to this lock and how it's necessary to have it? We're doing this lock for both orchestrators (Passthrough and Wintermute) so it's very important any future orchestrators follow the same pattern

Reply:
Valid point, realized most probably every orchestrator needs that lock. So moved the lock from orchestrators to the attack network: https://github.com/onflow/flow-go/pull/2599/commits/03aab4d6c7cb031f4d5921d2328340761030cc01. In this way, no further cautions needed from test developer on making the attack network --> orchestrator interaction thread-safe.

Reply:
Added further explanations also here: https://github.com/onflow/flow-go/pull/2599/commits/8c9afe7ed2e11549c70f4cf19401cdd8a9597caa